lib/rucio/daemons/rsedecommissioner/profiles/generic.py

```diff
@@ -0,0 +1,512 @@
+        tombstone = replica['tombstone']
+        
+        updated = False
+        
+        # If the replica fails any of the following three conditions, reaper
+        # won't pick it up
+        reap_pending = True
+        
+        # 1. Condition on the replica state & update time
+        in_stable_state = (state in [ReplicaState.AVAILABLE,
+                                     ReplicaState.UNAVAILABLE,
+                                     ReplicaState.BAD])
+        deleting_too_long = (state == ReplicaState.BEING_DELETED
+                             and updated_at < (datetime.utcnow()
+                                               - datetime.timedelta(seconds=600)))
+
+        if not (in_stable_state or deleting_too_long):
+            # 1.1 If COPYING without a valid request
+            # -> Update the state to UNAVAILABLE
+            if state == ReplicaState.COPYING:
+                # Should we allow requests in certain states? Depends on the
+                # details of the request state machine
+                count_stmt = select(
+                    func.count()
+                ).select_from(
+                    select(
+                        models.Request.id
+                    ).filter(
+                        models.Request.scope == scope,
+                        models.Request.name == name,
+                        models.Request.dest_rse_id == rse['id']
+                    ).subquery()
+                )
+
+                num_requests = session.execute(count_stmt).scalar_one()
+
+                if num_requests == 0:
+                    logger(logging.INFO,
+                           '(%s) Replica %s:%s is in COPYING state without a'
+                           ' valid request. Changing state to UNAVAILABLE.',
+                           rse['rse'], scope, name)
+                        
+                    replica.state = ReplicaState.UNAVAILABLE
+                    updated = True
+                    
+            reap_pending = False
+
+        # 2. Condition on the lock count
+        if lock_cnt != 0:
+            # 2.1 No actual lock -> Reset the lock count
+            count_stmt = select(
+                func.count()
+            ).select_from(
+                select(
+                    models.ReplicaLock.rule_id
+                ).filter(
+                    models.ReplicaLock.scope == scope,
+                    models.ReplicaLock.name == name,
+                    models.ReplicaLock.rse_id == rse['id']
+                ).subquery()
+            )
+
+            actual_lock_count = session.execute(count_stmt).scalar_one()
+            
+            if actual_lock_count == 0:
+                logger(logging.INFO,
+                       '(%s) Replica %s:%s has lock count %s but zero actual'
+                       ' locks. Changing lock_cnt to 0.',
+                       rse['rse'], scope, name, lock_cnt)
+                    
+                replica.lock_cnt = 0
```

Comment:
Unfortunately, there is no core function for this one and I think it would be inadvisable to add it. Hopefully the core issue behind it has since been fixed. Maybe we should convert this action to a warning-level log message instead?